nipap/nipap/rest.py

```diff
@@ -0,0 +1,240 @@
+            result["added"] = str(result.get("added"))
+            result["expires"] = str(result.get("expires"))
+            result["last_modified"] = str(result.get("last_modified"))
+            result["used_addresses"] = str(result.get("used_addresses"))
+
+            result = jsonify(result)
+            return result
+        except (AuthError, NipapError) as exc:
+            self.logger.debug(unicode(exc))
+            abort(500, error={"code": 500, "message": str(exc)})
+
+    @requires_auth
+    def remove_prefix(self, args):
+        return args
+
+
+    def delete(self):
+        combined_args = self.combine_request_args(False)
+        args = self.remove_prefix(combined_args)
+
+        try:
+            result = self.nip.remove_prefix(args.get('auth'), args.get('prefix'))
+            return jsonify(args.get('prefix'))
+        except (AuthError, NipapError) as exc:
+            self.logger.debug(unicode(exc))
+            abort(500, error={"code": 500, "message": exc}) 
+
+
+    def combine_request_args(self, isPost):
+        args = {}
+        
+        request_authoritative_source = request.headers.get('NIPAP-Authoritative-Source')
+        request_nipap_username = request.headers.get('NIPAP-Username')
+        request_nipap_fullname = request.headers.get('NIPAP-Full-Name')
+
+        request_body = request.json
+        request_queries = request.args.to_dict()
+
+        if request_authoritative_source:
+            authoritative_source = { 'auth':{'authoritative_source': request_authoritative_source}}
+            args.update(authoritative_source)
+
+            if request_nipap_username:
+                args['auth'].update({'username': request_nipap_username})
+            
+            if request_nipap_fullname:
+                args['auth'].update({'full_name': request_nipap_fullname})
+        
+        if request_queries and isPost:
+            temp_args = {}
+            if request_queries.get("fromPoolName"):
+                from_pool_name = {"from-pool": {"name": str(request_queries.get("fromPoolName"))}}
+                family = {"family": request_queries.get("family")}
+                prefix_length = {"prefix_length": int(request_queries.get("prefixLength"))}
+                temp_args.update(from_pool_name)
+                temp_args.update(family)
+                temp_args.update(prefix_length)
+            else:
+                from_prefix = {"from-prefix": [request_queries.get("fromPrefix")]}
+                prefix_length = {"prefix_length": request_queries.get("prefixLength")}
+                temp_args.update(from_prefix)
+                temp_args.update(prefix_length)
+
+            args.update({'args': temp_args})
+
+        elif request_queries and not isPost:
+            args.update({'prefix': request_queries})
+        
+        if request_body:
+            args.update(request_body)
+
+        return args
+
+
+if __name__ == '__main__':
```

Comment:
This block we can remove. It won't work as it stands today and I don't see when we'd like to use it.